### ISSUE TITLE
Include tag is not parsed for a resource type parameter

### ISSUE THREAD

**Issue by vovan-st** [raml-org/raml-java-parser#103]
The parser does not parse and include to external file with example if it is passed as parameter to resource type. Therefore, example of refactoring RAML file with using includes described in the [RAML tutorial](http://raml.org/developers/raml-200-tutorial#includes)  does not work:

```
/songs:
  type:
    collection:
      exampleCollection: !include jukebox-include-songs.sample
      exampleItem: !include jukebox-include-song-new.sample
```
Node tag (that is !include in this case) is ignored and the node is not resolved with IncludeResolver, but  only string value of the tag (that is a file path or name) is used while extracting resource references and cloning nodes. And later when resolving it using parameters it just set that string as parameter value.
Could you please help to resolve the issue with?

**Comment by svacas** [comment]
looks like issue #48, that was fixed in 0.8.12, did you check it with that version or greater?

**Comment by vovan-st** [comment]
@svacas Thanks a lot for your help! I did not pay attention which version of the parser is used in the project (0.8.6). It works with a newer version. Thanks again!
